symengine/polys.h

```diff
@@ -0,0 +1,70 @@
+/**
+ *  \file polys.h
+ *  Class for sparse Polynomial: Polynomial
+ *
+ **/
+#ifndef SYMENGINE_POLYS_H
+#define SYMENGINE_POLYS_H
+
+#include <symengine/basic.h>
+#include <symengine/dict.h>
+
+namespace SymEngine {
+//! Polynomial Class
+class Polynomial : public Basic{
+public:
+    //! `vars_` : Variables of the Polynomial
+    //! `polys_set_` : holds the Polynomial
+    vec_symbol vars_;
+    hash_set polys_set_;
```

Comment:
@certik or, to put it another way, it's my problem to update both ``hash_set.hpp`` and ``hash_set_fwd.hpp``. But I don't have strong feelings about this, your way works too.

More generally, I see from the Boost source code that it seems common practice to include in the public API just the ``_fwd`` headers, and now I start to see why :)